### PR TITLE
Vagrant

### PR DESCRIPTION
**Pull request by Zmoki**
I added Vagrantfile for use [Vagrant](https://www.vagrantup.com/) for start virtual machine with Ubuntu and preinstalled enviroment.

### CODE REVIEW

File: README.markdown [hunk 1]
```diff
@@ -74,40 +74,69 @@ Add following code to your document's `<head>`.
 
 ## Building Your Own
 
-You need a working Ruby 2.0.0 environment
-with [Bundler](http://bundler.io).
+Clone the repository.
+
+### Enviroment
+
+You need a working Ruby ^2.0.0 environment with [Bundler](http://bundler.io).
```
Comment (homm) [inline README.markdown]: Ruby 2.0.0 and above

File: Vagrantfile [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        sudo apt-get install -y build-essential ruby2.3-dev zlib1g-dev
+        sudo apt-get install -y git mercurial
+    SHELL
+
+    config.vm.provision "install", type: "shell", inline: <<-SHELL
+        cd /vagrant
+        bundle install
+        cd /vagrant/test/dummy
+        bundle install
+        bundle update
```
Comment (homm) [inline Vagrantfile]: This is extra for every install. Update will bump all packages versions, but only in `test/dummy` project.

File: README.markdown [hunk 3]
```diff
@@ -74,40 +74,69 @@ Add following code to your document's `<head>`.
...
+
+#### Vagrant
+
+If you don't want to raise the environment on your machine, you can use the [Vagrant](https://www.vagrantup.com/).
```
Comment (homm) [inline README.markdown]: can use ~~the~~ Vagrant

File: Vagrantfile (hunk 2 above)
Comment (Zmoki) [inline Vagrantfile]: @homm what do you mean? These provision shell script will be run only **on the first** "vagrant up" (https://www.vagrantup.com/docs/provisioning/). Maybe just 'bundle update' is extra?

File: Vagrantfile [hunk 4]
```diff
@@ -0,0 +1,24 @@
+Vagrant.configure(2) do |config|
+    config.vm.box = "ubuntu/trusty64"
+
+    config.vm.network "forwarded_port", guest: 3000, host: 3000
+
+    config.vm.provision "preinstall", type: "shell", inline: <<-SHELL
+        sudo apt-get update
+        sudo apt-get install -y software-properties-common
```
Comment (homm) [inline Vagrantfile]: `sudo apt-get install -y software-properties-common git mercurial build-essential zlib1g-dev`

File: Vagrantfile [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+        sudo apt-add-repository -y ppa:brightbox/ruby-ng
+        sudo apt-get update
+        sudo apt-get install -y ruby2.3
```
Comment (homm) [inline Vagrantfile]: `sudo apt-get install -y ruby2.3 ruby2.3-dev`

File: README.markdown [hunk 6]
```diff
@@ -74,40 +74,69 @@ Add following code to your document's `<head>`.
...
+
+Vagrant will install Ubuntu 14.04 x64 on VM, install all dependencies (like a ruby), install dependencies for project and tests (bundle install).
+
+After all, run:
+
+    vagrant ssh
+    
+Inside go to folder `/vagrant`.
+
+#### Without Vagrant: install dependencies
```
Comment (homm) [inline README.markdown]: Local environment (Without Vagrant)

File: README.markdown [hunk 7]
```diff
@@ -74,40 +74,69 @@ Add following code to your document's `<head>`.
...
+
+Inside folder run
+
+    bundle install
```
Comment (homm) [inline README.markdown]:
```
bundle install
cd ./test/dummy
bundle install
```

File: README.markdown [hunk 8]
```diff
@@ -74,40 +74,69 @@ Add following code to your document's `<head>`.
...
 
+### Development
 
-## Development
-
-Clone the repository, and go to `test/dummy/`. There is a simple Rails app. Run it:
+Go to `test/dummy/`. There is a simple Rails app. Run it:
 
-    bundle install
-    bundle update
```
Comment (homm) [inline README.markdown]: `cd ./test/dummy`

File: README.markdown [hunk 9]
```diff
@@ -74,40 +74,69 @@ Add following code to your document's `<head>`.
...
     bundle exec rails server
     
-Open http://0.0.0.0:3000/ . Follow any link. 
+Open http://0.0.0.0:3000/ (or http://127.0.0.1:3000 if you use Vagrant). Follow any link.
```
Comment (homm) [inline README.markdown]: if you are using

File: README.markdown [hunk 10]
```diff
@@ -74,40 +74,69 @@ Add following code to your document's `<head>`.
...
 
 
-## Testing
+### Testing
 
 [Jasminerice](https://github.com/bradphelan/jasminerice) 
 installed under the `test/dummy/` Rails app.
 
-To run tests in your browser go to http://0.0.0.0:3000/jasmine.
+To run tests in your browser go to http://0.0.0.0:3000/jasmine (or http://127.0.0.1:3000/jasmine if you use Vagrant).
```
Comment (homm) [inline README.markdown]: To run tests in your browser go to http://127.0.0.1:3000/jasmine.

File: README.markdown (hunk 9 above)
Comment (homm) [inline README.markdown]: Open http://127.0.0.1:3000/. Follow any link.

**Comment by homm** [comment]
:+1:

**Pull request merged**